simalign/simalign.py

```diff
@@ -59,12 +62,10 @@ def __init__(self, model="bert-base-multilingual-cased", device=torch.device('cp
 
 	def get_embed_list(self, sent_pair):
 		if self.emb_model is not None:
-			sent_ids = [self.tokenizer.convert_tokens_to_ids(x) for x in sent_pair]
-			inputs = [self.tokenizer.prepare_for_model(sent, return_token_type_ids=True, return_tensors='pt')['input_ids'] for sent in sent_ids]
+			inputs = self.tokenizer(sent_pair, is_pretokenized=True, padding=True, truncation=True, return_tensors="pt")
+			outputs = self.emb_model(**inputs)[2][self.layer]
 
-			outputs = [self.emb_model(in_ids.to(self.device)) for in_ids in inputs]
-			# use vectors from layer 8
-			vectors = [x[2][self.layer].cpu().detach().numpy()[0][1:-1] for x in outputs]
+			vectors = F.normalize(outputs[:, 1:-1, :], dim=2)

```

Comment:
It is needed for later changes. This way, instead of using `get_similarity()`, the `matmul()` function can be used. It can be performed on several sentence pairs together which makes it much faster.